lib/Sema/ConstraintLocator.h

```diff
@@ -319,6 +323,11 @@ class ConstraintLocator : public llvm::FoldingSetNode {
       assert(getAssociatedType() == decl);
     }
 
+    PathElement(const swift::Requirement &req, unsigned pos)
```

Comment:
`req` isn't actually used here; can you instead add a `static PathElement getTypeRequirement(unsigned position)`?

Reply:
Will do, thanks!